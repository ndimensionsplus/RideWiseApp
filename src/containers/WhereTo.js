import React from 'react';
import {
  Platform, 
  Text, 
  View, 
  StyleSheet, 
  TouchableOpacity, 
  SafeAreaView 
} from 'react-native';

class WhereTo extends React.Component {
  
  goBack = () => {
    this.props.navigation.navigate('Home')
  }

  render() {
    return (
      <SafeAreaView style={{flex: 1, backgroundColor: '#ebebeb'}}>
        <View style={styles.container}>
          <View>
            <TouchableOpacity onPress={this.goBack}>
              <Text style={{ fontFamily: Platform.OS === 'android' ? 'fontawesome' : 'Font Awesome 5 Free', fontSize: 20 }}>&#xf104;</Text>
            </TouchableOpacity>
          </View>
          <View style={{flex: 1, paddingTop: 20}}>
            <Text>Where To</Text>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 20,
  },
});

export default WhereTo;
