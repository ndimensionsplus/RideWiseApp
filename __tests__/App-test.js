/**
 * @format
 */

import 'react-native';
import React from 'react';
import { Home, WhereTo, Settings } from '../src/containers';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("App Test Cases", () => {

  const nav = {
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    goBack: jest.fn()
  };

  it('Home Page Renders Correctly', () => {
    const tree = renderer.create(<Home navigation={nav} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Where To Page Renders Correctly', () => {
    const tree = renderer.create(<WhereTo navigation={nav} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Settings Page Renders Correctly', () => {
    const tree = renderer.create(<Settings navigation={nav} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
