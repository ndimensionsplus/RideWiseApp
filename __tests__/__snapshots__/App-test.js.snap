// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Cases Home Page Renders Correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "height": 20,
          "left": 30,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 50,
          "width": 20,
          "zIndex": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Font Awesome 5 Free",
            "fontSize": 20,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "margin": 20,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        }
      }
    >
      <View>
        <Text>
          Where To
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`App Test Cases Settings Page Renders Correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "height": 20,
        "left": 30,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 50,
        "width": 20,
        "zIndex": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Font Awesome 5 Free",
          "fontSize": 20,
        }
      }
    >
      
    </Text>
  </View>
  <Text>
    Settings!
  </Text>
</View>
`;

exports[`App Test Cases Where To Page Renders Correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Font Awesome 5 Free",
              "fontSize": 20,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 20,
        }
      }
    >
      <Text>
        Where To
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
